source/server-discovery-and-monitoring/tests/unified/closeInUse-pool-clear.yml

```diff
@@ -0,0 +1,311 @@
+        arguments:
+          document:
+            _id: 1
+      - name: runOnThread
+        object: testRunner
+        arguments:
+          thread: *thread1
+          operation:
+            name: find
+            object: *collection
+            arguments:
+              filter:
+                $where : sleep(5000) || true
+            expectError:
+              isError: true
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *setupClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode:
+              times: 2
+            data:
+              failCommands:
+                - hello, isMaster
+              blockConnection:  true,
+              blockTimeMS: 1500,                
+              appName: closeInUseRetryable        
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            commandFailedEvent: {}
+          count: 1
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            poolClearedEvent: { closeInUseConnections: true }
+          count: 1
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            connectionCheckedInEvent: { }
+          count: 3
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            connectionClosed: { reason: stale }
+          count: 1
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            commandFailedEvent: { }
+          count: 1
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            poolClearedEvent: { }
+          count: 1
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            connectionClosedEvent: { }
+          count: 1
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            connectionCheckedOutEvent: { }
+          count: 3
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            connectionCheckedInEvent: { }
+          count: 3
+
+    expectEvents:
+      - client: *client
+        eventType: command
+        events:
+          - commandStartedEvent:
+              commandName: insert
+          - commandStartedEvent:
+              commandName: find
+          - commandFailedEvent:
+              commandName: find
+          - commandStartedEvent:
+              commandName: find
+    outcome:
+      - collectionName: *collectionName
+        databaseName: *databaseName
+        documents:
+          - _id: 1
```

Comment:
discussed offline, more reliable scenario is:
* run an insert to get primary discovered
* enable fp on insert, hello with high blockTimeMS
* execute insert
The monitor should time out after connectTimeMS + heartbeatFrequencyMS (set them so that blockTimeMS is higher) and cancel the insert, and it should retry once the server is rediscovered